From 9f7a9e36a59ee22644202e590117f780ed58110e Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Tue, 4 Jun 2019 21:42:40 +0300
Subject: [PATCH 001/211] ARM64: dts: Remove conflicting with mainline V3M and
 V3H device tree parts

This removes V3M and V3H device tree parts which conflict
with the mainline renesas-devel-20190603-v5.2-rc3 tag.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/Makefile            |  10 -
 arch/arm64/boot/dts/renesas/r8a77970-eagle.dts  | 356 ------------------------
 arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts  |  71 -----
 arch/arm64/boot/dts/renesas/r8a77980-condor.dts |  40 ---
 arch/arm64/boot/dts/renesas/r8a77980-v3hsk.dts  |  41 ---
 5 files changed, 518 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index f7f8638..294d38a 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -33,15 +33,5 @@ dtb-$(CONFIG_ARCH_R8A7795) += r8a7795-h3ulcb-vb2.dtb r8a7795-es1-h3ulcb-vb2.dtb
 dtb-$(CONFIG_ARCH_R8A7795) += r8a7795-h3ulcb-vb2.1.dtb r8a7795-h3ulcb-4x2g-vb2.1.dtb
 dtb-$(CONFIG_ARCH_R8A7795) += r8a7795-h3ulcb-vbm.dtb r8a7795-es1-h3ulcb-vbm.dtb r8a7795-h3ulcb-4x2g-vbm.dtb
 dtb-$(CONFIG_ARCH_R8A77965) += r8a77965-m3nulcb-kf.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-eagle-function.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-es1-eagle.dtb r8a77970-es1-eagle-function.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-es1-v3msk.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-v3msk-view.dtb r8a77970-es1-v3msk-view.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-v3msk-kf.dtb r8a77970-es1-v3msk-kf.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-v3msk-vbm.dtb r8a77970-es1-v3msk-vbm.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-v3msk-vbm-v2.dtb r8a77970-es1-v3msk-vbm-v2.dtb
-#dtb-$(CONFIG_ARCH_R8A77970) += r8a77970-v3mzf.dtb
-#dtb-$(CONFIG_ARCH_R8A77980) += r8a77980-v3hsk-vbm.dtb
-#dtb-$(CONFIG_ARCH_R8A77980) += r8a77980-v3hsk-vbm-v2.dtb
 
 always		:= $(dtb-y)
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts b/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts
index 6338ab3..b6d5332 100644
--- a/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77970-eagle.dts
@@ -8,7 +8,6 @@
 
 /dts-v1/;
 #include "r8a77970.dtsi"
-#include <dt-bindings/gpio/gpio.h>
 
 / {
 	model = "Renesas Eagle board based on r8a77970";
@@ -30,58 +29,6 @@
 		reg = <0x0 0x48000000 0x0 0x38000000>;
 	};
 
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress {
-			no-map;
-			reg = <0x00000000 0x6c000000 0x0 0x03000000>;
-		};
-
-		/* global autoconfigured region for contiguous allocations */
-		linux,cma {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x6f000000 0x0 0x10000000>;
-			linux,cma-default;
-		};
-
-		/* device specific region for contiguous allocations */
-		linux,multimedia {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x7f000000 0x0 0x01000000>;
-		};
-	};
-
-	mmngr {
-		compatible = "renesas,mmngr";
-		memory-region = <&lossy_decompress>;
-	};
-
-	mmngrbuf {
-		compatible = "renesas,mmngrbuf";
-	};
-
-	vspm_if {
-		compatible = "renesas,vspm_if";
-	};
-
-	dclkin_p0: clock-out0 {
-		compatible = "fixed-clock";
-		#clock-cells = <0>;
-		clock-frequency = <148500000>;
-	};
-
-	msiof_ref_clk: msiof-ref-clock {
-		compatible = "fixed-clock";
-		#clock-cells = <0>;
-		clock-frequency = <66666666>;
-	};
-
 	hdmi-out {
 		compatible = "hdmi-connector";
 		type = "a";
@@ -155,40 +102,6 @@
 	};
 };
 
-&csi40 {
-	status = "okay";
-
-	virtual,channel {
-		csi2_vc0 {
-			data,type = "ycbcr422";
-			receive,vc = <0>;
-		};
-		csi2_vc1 {
-			data,type = "ycbcr422";
-			receive,vc = <1>;
-		};
-		csi2_vc2 {
-			data,type = "ycbcr422";
-			receive,vc = <2>;
-		};
-		csi2_vc3 {
-			data,type = "ycbcr422";
-			receive,vc = <3>;
-		};
-	};
-
-	port {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		csi2_40_ep: endpoint {
-			clock-lanes = <0>;
-			data-lanes = <1 2 3 4>;
-			csi-rate = <300>;
-		};
-	};
-};
-
 &extal_clk {
 	clock-frequency = <16666666>;
 };
@@ -242,158 +155,6 @@
 			};
 		};
 	};
-
-	pmic@5A {
-		compatible = "dlg,da9063";
-		reg = <0x5A>;
-		interrupt-parent = <&intc_ex>;
-		interrupts = <0 IRQ_TYPE_LEVEL_LOW>;
-		interrupt-controller;
-
-		rtc {
-			compatible = "dlg,da9063-rtc";
-		};
-
-		wdt {
-			compatible = "dlg,da9063-watchdog";
-		};
-
-		regulators {
-			DA9063_LDO11: bmem {
-				regulator-name = "bmem";
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-				regulator-boot-on;
-			};
-		};
-
-		onkey {
-			compatible = "dlg,da9063-onkey";
-		};
-	};
-};
-
-&i2c3 {
-	pinctrl-0 = <&i2c3_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-	clock-frequency = <400000>;
-
-	ov106xx@0 {
-		compatible = "ovti,ov106xx";
-		reg = <0x60>;
-
-		port@0 {
-			ov106xx_in0: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&vin0ep0>;
-			};
-		};
-		port@1 {
-			ov106xx_max9286_des0ep0: endpoint@0 {
-				remote-endpoint = <&max9286_des0ep0>;
-			};
-		};
-	};
-
-	ov106xx@1 {
-		compatible = "ovti,ov106xx";
-		reg = <0x61>;
-
-		port@0 {
-			ov106xx_in1: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&vin1ep0>;
-			};
-		};
-		port@1 {
-			ov106xx_max9286_des0ep1: endpoint@0 {
-				remote-endpoint = <&max9286_des0ep1>;
-			};
-		};
-	};
-
-	ov106xx@2 {
-		compatible = "ovti,ov106xx";
-		reg = <0x62>;
-
-		port@0 {
-			ov106xx_in2: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&vin2ep0>;
-			};
-		};
-		port@1 {
-			ov106xx_max9286_des0ep2: endpoint@0 {
-				remote-endpoint = <&max9286_des0ep2>;
-			};
-		};
-	};
-
-	ov106xx@3 {
-		compatible = "ovti,ov106xx";
-		reg = <0x63>;
-
-		port@0 {
-			ov106xx_in3: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&vin3ep0>;
-			};
-		};
-		port@1 {
-			ov106xx_des0ep3: endpoint {
-				remote-endpoint = <&max9286_des0ep3>;
-			};
-		};
-	};
-
-	max9286@0 {
-		compatible = "maxim,max9286";
-		reg = <0x48>;
-		gpios = <&io_expander 0 GPIO_ACTIVE_LOW>; /* CSI0 DE_PDn */
-		maxim,gpio0 = <0>;
-		maxim,sensor_delay = <100>;
-		maxim,links = <4>;
-		maxim,lanes = <4>;
-		maxim,resetb-gpio = <1>;
-		maxim,fsync-mode = "automatic";
-		maxim,timeout = <100>;
-
-		port@0 {
-			max9286_des0ep0: endpoint@0 {
-				max9271-addr = <0x50>;
-				dvp-order = <1>;
-				remote-endpoint = <&ov106xx_in0>;
-			};
-			max9286_des0ep1: endpoint@1 {
-				max9271-addr = <0x51>;
-				dvp-order = <1>;
-				remote-endpoint = <&ov106xx_in1>;
-			};
-			max9286_des0ep2: endpoint@2 {
-				max9271-addr = <0x52>;
-				dvp-order = <1>;
-				remote-endpoint = <&ov106xx_in2>;
-			};
-			max9286_des0ep3: endpoint@3 {
-				max9271-addr = <0x53>;
-				dvp-order = <1>;
-				remote-endpoint = <&ov106xx_in3>;
-			};
-		};
-		port@1 {
-			max9286_csi0ep0: endpoint {
-				csi-rate = <700>;
-				remote-endpoint = <&csi2_40_ep>;
-			};
-		};
-	};
 };
 
 &pfc {
@@ -412,11 +173,6 @@
 		function = "i2c0";
 	};
 
-	i2c3_pins: i2c3 {
-		groups = "i2c3";
-		function = "i2c3";
-	};
-
 	scif0_pins: scif0 {
 		groups = "scif0_data";
 		function = "scif0";
@@ -435,118 +191,6 @@
 	status = "okay";
 };
 
-&vin0 {
-	status = "okay";
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		port@0 {
-			vin0ep0: endpoint {
-				csi,select = "csi40";
-				virtual,channel = <0>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&ov106xx_in0>;
-			};
-		};
-		port@1 {
-			csi0ep0: endpoint {
-				remote-endpoint = <&csi2_40_ep>;
-			};
-		};
-		port@2 {
-			vin0_max9286_des0ep0: endpoint@0 {
-				remote-endpoint = <&max9286_des0ep0>;
-			};
-		};
-	};
-};
-
-&vin1 {
-	status = "okay";
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		port@0 {
-			vin1ep0: endpoint {
-				csi,select = "csi40";
-				virtual,channel = <1>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&ov106xx_in1>;
-			};
-		};
-		port@1 {
-			csi0ep1: endpoint {
-				remote-endpoint = <&csi2_40_ep>;
-			};
-		};
-		port@2 {
-			vin1_max9286_des0ep1: endpoint@0 {
-				remote-endpoint = <&max9286_des0ep1>;
-			};
-		};
-	};
-};
-
-&vin2 {
-	status = "okay";
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		port@0 {
-			vin2ep0: endpoint {
-				csi,select = "csi40";
-				virtual,channel = <2>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&ov106xx_in2>;
-			};
-		};
-		port@1 {
-			csi0ep2: endpoint {
-				remote-endpoint = <&csi2_40_ep>;
-			};
-		};
-		port@2 {
-			vin2_max9286_des0ep2: endpoint@0 {
-				remote-endpoint = <&max9286_des0ep2>;
-			};
-		};
-	};
-};
-
-&vin3 {
-	status = "okay";
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		port@0 {
-			vin3ep0: endpoint {
-				csi,select = "csi40";
-				virtual,channel = <3>;
-				data-lanes = <1 2 3 4>;
-				remote-endpoint = <&ov106xx_in3>;
-			};
-		};
-		port@1 {
-			csi0ep3: endpoint {
-				remote-endpoint = <&csi2_40_ep>;
-			};
-		};
-		port@2 {
-			vin3_max9286_des0ep3: endpoint@0 {
-				remote-endpoint = <&max9286_des0ep3>;
-			};
-		};
-	};
-};
-
 &du {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts b/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts
index 2b614e4..8eac8ca 100644
--- a/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77970-v3msk.dts
@@ -27,46 +27,6 @@
 		reg = <0x0 0x48000000 0x0 0x38000000>;
 	};
 
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress {
-			no-map;
-			reg = <0x00000000 0x6c000000 0x0 0x03000000>;
-		};
-
-		/* global autoconfigured region for contiguous allocations */
-		linux,cma {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x6f000000 0x0 0x10000000>;
-			linux,cma-default;
-		};
-
-		/* device specific region for contiguous allocations */
-		linux,multimedia {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x7f000000 0x0 0x01000000>;
-		};
-	};
-
-	mmngr {
-		compatible = "renesas,mmngr";
-		memory-region = <&lossy_decompress>;
-	};
-
-	mmngrbuf {
-		compatible = "renesas,mmngrbuf";
-	};
-
-	vspm_if {
-		compatible = "renesas,vspm_if";
-	};
-
 	osc5_clk: osc5-clock {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
@@ -218,37 +178,6 @@
 			};
 		};
 	};
-
-	pmic@5A {
-		compatible = "dlg,da9063";
-		reg = <0x5A>;
-		interrupt-parent = <&intc_ex>;
-		interrupts = <0 IRQ_TYPE_LEVEL_LOW>;
-		interrupt-controller;
-
-		rtc {
-			compatible = "dlg,da9063-rtc";
-		};
-
-		wdt {
-			compatible = "dlg,da9063-watchdog";
-		};
-
-		regulators {
-			DA9063_LDO11: bmem {
-				regulator-name = "bmem";
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-				regulator-boot-on;
-			};
-		};
-
-		onkey {
-			compatible = "dlg,da9063-onkey";
-		};
-	};
-
 };
 
 &lvds0 {
diff --git a/arch/arm64/boot/dts/renesas/r8a77980-condor.dts b/arch/arm64/boot/dts/renesas/r8a77980-condor.dts
index e270cc0..0b93a7d 100644
--- a/arch/arm64/boot/dts/renesas/r8a77980-condor.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77980-condor.dts
@@ -28,46 +28,6 @@
 		reg = <0 0x48000000 0 0x78000000>;
 	};
 
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress {
-			no-map;
-			reg = <0x00000000 0x6c000000 0x0 0x03000000>;
-		};
-
-		/* global autoconfigured region for contiguous allocations */
-		linux,cma {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x6f000000 0x0 0x10000000>;
-			linux,cma-default;
-		};
-
-		/* device specific region for contiguous allocations */
-		linux,multimedia {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x7f000000 0x0 0x01000000>;
-		};
-	};
-
-	mmngr {
-		compatible = "renesas,mmngr";
-		memory-region = <&lossy_decompress>;
-	};
-
-	mmngrbuf {
-		compatible = "renesas,mmngrbuf";
-	};
-
-	vspm_if {
-		compatible = "renesas,vspm_if";
-	};
-
 	d3_3v: regulator-0 {
 		compatible = "regulator-fixed";
 		regulator-name = "D3.3V";
diff --git a/arch/arm64/boot/dts/renesas/r8a77980-v3hsk.dts b/arch/arm64/boot/dts/renesas/r8a77980-v3hsk.dts
index dff7d02..c968099 100644
--- a/arch/arm64/boot/dts/renesas/r8a77980-v3hsk.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77980-v3hsk.dts
@@ -26,47 +26,6 @@
 		/* first 128MB is reserved for secure area. */
 		reg = <0 0x48000000 0 0x78000000>;
 	};
-
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress {
-			no-map;
-			reg = <0x00000000 0x6c000000 0x0 0x03000000>;
-		};
-
-		/* global autoconfigured region for contiguous allocations */
-		linux,cma {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x6f000000 0x0 0x10000000>;
-			linux,cma-default;
-		};
-
-		/* device specific region for contiguous allocations */
-		linux,multimedia {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x7f000000 0x0 0x01000000>;
-		};
-	};
-
-	mmngr {
-		compatible = "renesas,mmngr";
-		memory-region = <&lossy_decompress>;
-	};
-
-	mmngrbuf {
-		compatible = "renesas,mmngrbuf";
-	};
-
-	vspm_if {
-		compatible = "renesas,vspm_if";
-	};
-
 };
 
 &extal_clk {
-- 
2.7.4

